astropy/coordinates/sites.py

```diff
@@ -105,8 +105,11 @@ def from_json(cls, jsondb):
             location = EarthLocation.from_geodetic(site_info.pop('longitude') * u.Unit(site_info.pop('longitude_unit')),
                                                    site_info.pop('latitude') * u.Unit(site_info.pop('latitude_unit')),
                                                    site_info.pop('elevation') * u.Unit(site_info.pop('elevation_unit')))
-            location.info.name = site_info.pop('name')
-            aliases = site_info.pop('aliases')
+            name = site_info.pop('name')
+            location.info.name = name
+            aliases = [alias for alias in site_info.pop('aliases') if alias]
+            if name not in aliases:
+                aliases.append(name)
```

Comment:
I wonder if it makes sense to raise an error or a warning here instead of silently dropping names? I know that this was no treated properly before, so while making it better, let's make is as good as we can!
I see two possible problems that may or may not turn up in the current sites.json file:
- An observatory is added twice and either name or aliases overlap.
- Two different observatories register the same alias or name. While that should not happen for big observatories (there is only one VLT) or long names ("Tautenburger Sternwarte"), I would not be surprised if the same three-letter acronym is used for different observatories in different parts of the world.

Either condition should not happen in a well curated sites.json, but (a) mistakes happen, (b) the user could supply their own sites.json and (c) it's very easy to just just an "else: raise XXX" in here. 

I suggest that a warning is sufficient, since sites.json is long and an error in observatory "SWO" should not stop me from getting coordinates for a totally unlates query like "Gemini/North".

Reply:
Raising warnings about duplicated names sounds like a good idea, but it should not be implemented until the `sites.json` file in `astropy-data` is cleaned from duplicates, otherwise the users would receive many warnings that they can't really do anything about. Removing the duplicates from `sites.json` would reduce the number of usable keys for anyone not using the bleeding edge version of `astropy`, so it doesn't seem it would be a good idea to do that quite yet.

When it comes to checking for duplicated names for different sites then it wouldn't be too difficult to add a test to `astropy-data` that would check for that. The output of `astropy.coordinates.EarthLocation.get_site_names()` shows that currently there aren't any duplicated labels across sites other than the empty string that this pull request takes care of.

Reply:
Could you open an issue over in astropy/astropy-data to remind about this?

Reply:
There is a change log now (thanks!) and I opened issue at astropy/astropy-data#126 . Please do a final review of this PR and approve if satisfactory. Thanks!